tests/integration-tests/tests/iam/test_iam.py

```diff
@@ -145,9 +145,16 @@ def test_s3_read_write_resource(
     cluster_config = pcluster_config_reader(bucket=bucket_name)
     cluster = clusters_factory(cluster_config)
 
-    config = configparser.ConfigParser()
-    config.read(cluster_config)
+    with open(cluster_config) as conf_file:
+        config = yaml.load(conf_file, Loader=yaml.SafeLoader)
 
     # Check S3 resources
-    check_s3_read_resource(region, cluster, config.get("cluster default", "s3_read_resource"))
-    check_s3_read_write_resource(region, cluster, config.get("cluster default", "s3_read_write_resource"))
+    check_s3_read_resource(region, cluster, _get_bucket_name(config, False))
+    check_s3_read_write_resource(region, cluster, _get_bucket_name(config, True))
+
+
+def _get_bucket_name(config, enable_write_access):
+    s3_access = config["HeadNode"]["Iam"]["S3Access"]
+    for access in s3_access:
```

Comment:
you can use an approach like:
```python
bucket_name = next(access["BucketName"] for access in s3_access if access["EnableWriteAccess"] == enable_write_access, None)
if not bucket_name:
   error...
```
to explicitly catch cases on which one of the two fields is not defined in the conf.

Reply:
Done